### ISSUE TITLE
multiple instances dont parallelize strata

### ISSUE THREAD

**Issue by gtristan** [devcurmudgeon/ybd#137]
I observed ybd build gtk2 and then build gtk3 (separate strata) one after the other today; while clearly they do not depend on eachother; later ybd happily parallelizes chunk builds within the NetworkManager-common stratum.

We can get more bang for our buck here :)

**Comment by devcurmudgeon** [comment]
I don't think this is true. ybd should randomise order for building strata:

https://github.com/devcurmudgeon/ybd/blob/master/ybd/assembly.py#L77

so one instance could get (say) gtk2 and another (say) gtk3. If multiple instances go at gtk2, and a component there becomes a bottleneck, in instance which is forced to wait should timeout and retry from the top, and (maybe after more retries) find it can make progress on gtk3

Please could you paste a build log? or point me at your definitions repo and i can try building it?

**Comment by gtristan** [comment]
I think you are right... to a certain extent at least...

I am sure I observed the build blocking on gtk2 before building gtk3 the other day, however today I noticed some X drivers being built in parallel with the gnome stratum.

Perhaps the gtk build itself was just intense (and parallel within itself) enough so as to starve the parent ybd script into not scheduling the following build ? I doubt this as much of the linker phases are io bound.

I will try to capture and save a build log which shows this blocking, however I now doubt it has anything to do with chunks vs. stratum, just that in some cases it seems to block (perhaps having ybd launch the builds at a lower priority than itself would help ?)
